server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/jsieve/delivery/SieveExecutor.java

```diff
@@ -141,7 +141,6 @@ protected void sieveMessage(MailAddress recipient, Mail aMail, Log log) throws M
             // seems very unfriendly.
             // So just log and store in INBOX
             log.error("Cannot evaluate Sieve script. Storing mail in user INBOX.", ex);
-            storeMessageInbox(recipient, aMail);
```

Comment:
This job would be done by LocalDelivery.

Because sieve failed, default action will then be taken as there is no instruction in mail attributes.